x-pack/plugins/dataset_quality/server/types/default_api_types.ts

```diff
@@ -8,16 +8,20 @@
 import * as t from 'io-ts';
 import { isoToEpochRt } from '@kbn/io-ts-utils';
 
-export const dataStreamTypesRt = t.partial({
-  type: t.union([
-    t.literal('logs'),
-    t.literal('metrics'),
-    t.literal('traces'),
-    t.literal('synthetics'),
-    t.literal('profiling'),
-  ]),
+export const dataStreamTypesRt = t.union([
+  t.literal('logs'),
+  t.literal('metrics'),
+  t.literal('traces'),
+  t.literal('synthetics'),
+  t.literal('profiling'),
+]);
```

Comment:
**nit:** the `io-ts`[docs](https://github.com/gcanti/io-ts/blob/master/index.md#union-of-string-literals) suggests using the `keyof` helper for string unions, would be great to follow their recommendation.

Reply:
Addressed in [3b1d170](https://github.com/elastic/kibana/pull/173087/commits/3b1d17073e86bada48c39eb8d33710f595aa9b8d).